File: order-service/src/main/java/com/paulopacifico/orderservice/messaging/api/OrderFailedEvent.java
package com.paulopacifico.orderservice.messaging.api;

import java.time.OffsetDateTime;
import java.util.UUID;

public record OrderFailedEvent(
        UUID eventId,
        Long orderId,
        String orderNumber,
        String skuCode,
        Integer reservedQuantity,
        String reason,
        OffsetDateTime occurredAt
) {
}

File: inventory-service/src/main/kotlin/com/paulopacifico/inventoryservice/messaging/api/OrderFailedEvent.kt
package com.paulopacifico.inventoryservice.messaging.api

import java.time.OffsetDateTime
import java.util.UUID

data class OrderFailedEvent(
    val eventId: UUID,
    val orderId: Long,
    val orderNumber: String,
    val skuCode: String,
    val reservedQuantity: Int,
    val reason: String,
    val occurredAt: OffsetDateTime,
)
